analyzer/ipsec/ipsec.spicy

```diff
@@ -286,6 +286,10 @@ type IPSecIKEv2_Payload = unit(inout shared_info: SharedInfo, message_id: uint32
 
 	header: IPSecIKE_Payload_Header { shared_info.next_payload = $$.next_payload; }
 
+	# Require a minimum payload_length of 4 to prevent triggering
+	# integer overflow errors in the switch statement below.
+	: bytes &size=0 &requires=(self.header.payload_length >= 4) if ( self.this_payload != PayloadType::NO_NEXT_PAYLOAD );
```

Comment:
I was wondering whether it would make sense to move this directly into a requirement on the definition of `payload_length`, e.g., that it needs to be `>=5), but at least from how this parser is structured it seems conceivable that it could also be less if we then do not extract any payload. I think this is 👍.

FYI, I filed zeek/spicy#1051 to improve the syntax for this.